test/sslapitest.c

```diff
@@ -9782,22 +9782,30 @@ static int test_configuration_of_groups(void)
         goto end;
 
     default_groups_len = ctx->ext.supported_groups_default_len;
-
+#if (!defined(OPENSSL_NO_EC) || !defined(OPENSSL_NO_DH))
     if (!TEST_size_t_gt(default_groups_len, 0)
         || !TEST_int_gt(SSL_CTX_set1_groups_list(ctx, "DEFAULT"), 0)
+#else
+    if (!TEST_size_t_eq(default_groups_len, 0)
+        || !TEST_int_eq(SSL_CTX_set1_groups_list(ctx, "DEFAULT"), 0)
+#endif
```

Comment:
I would not add these checks as in future they might be misleading, i.e. when we will have some additional groups such as ML-KEM. I.e., just keep the equivalency check.